### DISCUSSION TITLE
Redundant step in Setup of TLS Server?

### DISCUSSION THREAD

**Discussion by AnyFlyGuy** [open-quantum-safe/openssl#344] [Q&A] [answered]
Hi,

first of all, thank you for your work, highly appreciated.
I'm also fairly new to the topic, so maybe I'm wrong. In your documentation at the point of setting up a TLS server:

>1. The server generates its key pair:
>    `apps/openssl genpkey -algorithm <SIG> -out <SIG>_srv.key`
>
>2. The server generates a certificate request and sends it the to CA:
>
>    `apps/openssl req -new -newkey <SIG> -keyout <SIG>_srv.key -out <SIG>_srv.csr -nodes -subj "/CN=oqstest server" -config apps/openssl.cnf`

I think the fist step is not necessary because with `-newkey <SIG> -keyout <SIG>_srv.key` in the second command you also generate a key. Respectively if you want to make the generation of the key more clear, maybe the second command doesn't need the `-newkey` command and reference the generated key with the `-key` command.

**Comment by baentsch** [comment, marked as answer]
> I think the fist step is not necessary

At first glance this observation seems right. Thanks for pointing this out. Would you want to suggest a documentation update via PR or shall we fix it (I'd probably simply drop "Step 1")?

**Reply by AnyFlyGuy** [reply]
I'm fine with you changing it whenever you have the time.

**Reply by baentsch** [reply]
Done in #345 . Thanks again for simplifying things for everyone!
